mgmt-vrf-design-doc.md

```diff
@@ -0,0 +1,313 @@
+Management VRF Design Document
+
+Introduction
+------------
+
+Management VRF is a subset of VRF (virtual routing tables and forwarding) and provides a separation between the out-of-band management network and the in-band data plane network. For all VRFs, the main routing table is the default table for all of the data plane switch ports. With management VRF, a second table, mgmt, is used for routing through the Ethernet ports of the switch. The following design for mgmt-vrf uses the l3mdev approach for implementating mgmt-vrf on SONiC.
+
+Requirements
+------------
+                                              Table - 1
+    | Req No | Description                                                                                                | Priority | Comments |
+    |--------|------------------------------------------------------------------------------------------------------------|----------|----------|
+    | 1      | Develop and implement a separate Management VRF that provide management plane and Data Plane Isolation     |          |          |
+    | 2      | Management VRF should be associated with a separate L3 Routing table and the management interface          |          |          |
+    | 3      | Management VRF and Default VRF should support IP services like Ping and traceroute  in its context         |          |          |
+    | 4      | Management VRF should provide the ability to be polled via SNMP both via default VRF and management VRF    |          |          |
+    | 5      | Management VRF will provide the ability to send SNMP traps both via the   management VRF and data network. |          |          |
+    | 6      | Enable DHCP services to run in Management VRF context                                                      |          |          |
+    |        | ˇ       Dhcpd                                                                                              |          |          |
+    |        | ˇ       Dhcrelay  Default VRF support                                                                     |          |          |
+    |        | ˇ       dhclient                                                                                           |          |          |

```

Comment:
it is unclear from the design doc that how dhclient can be running in management vrf context and what is the implication for that.

Reply:
Will update.